src/sdk/flyover.test.ts

```diff
@@ -952,4 +963,116 @@ describe('Flyover object should', () => {
         .rejects.toThrow('Provider API base URL is not secure. Please enable insecure connections on Flyover configuration')
     })
   })
+
+  describe('hashPeginQuote method should', () => {
+    const MOCK_HASH = 'mocked-hash-value'
+
+    const mockLiquidityBridgeContract: LiquidityBridgeContract = {
+      hashPeginQuote: jest.fn()
+    } as unknown as LiquidityBridgeContract
+
+    beforeEach(() => {
+      jest.clearAllMocks()
+
+      jest.spyOn(mockLiquidityBridgeContract, 'hashPeginQuote').mockImplementation(async () => Promise.resolve(MOCK_HASH))
+      ;(flyover as any).liquidityBridgeContract = mockLiquidityBridgeContract
+    })
+
+    test('call liquidityBridgeContract.hashPeginQuote with the correct quote', async () => {
+      await flyover.connectToRsk(rskConnectionMock)
+
+      const result = await flyover.hashPeginQuote(quoteMock)
+
+      expect(mockLiquidityBridgeContract.hashPeginQuote).toHaveBeenCalledWith(quoteMock)
+      expect(result).toBe(MOCK_HASH)
+    })
+
+    test('return the hash computed by the LBC contract', async () => {
+      await flyover.connectToRsk(rskConnectionMock)
+      jest.spyOn(LiquidityBridgeContract.prototype, 'hashPeginQuote').mockResolvedValue(MOCK_HASH)
+
+      const result = await flyover.hashPeginQuote(quoteMock)
+
+      expect(result).toBe(MOCK_HASH)
+    })
+
+    test('throw error if not connected to RSK', async () => {
+      await expect(flyover.hashPeginQuote(quoteMock)).rejects.toThrow('Not connected to RSK')
+    })
+
+    test('create LBC instance if not created before', async () => {
+      flyover = new Flyover({
+        network: FAKE_NETWORK,
+        allowInsecureConnections: true,
+        captchaTokenResolver: async () => Promise.resolve('')
+      })
+
+      expect(flyover).not.toHaveProperty('liquidityBridgeContract')
+
+      await flyover.connectToRsk(rskConnectionMock)
+
+      // The result of this call is not important, we just want to check that the LBC instance is created
```

Comment:
then maybe you could rewrite as await expect(fn()).resolves.not.toThrow()

Reply:
It will actually throw an error because it will create a real LBC instance and use it.